misc/python/materialize/mzbuild.py

```diff
@@ -528,6 +530,14 @@ def inputs(self, transitive: bool = False) -> Set[str]:
                 repository.
         """
         paths = set(git.expand_globs(self.image.rd.root, f"{self.image.path}/**"))
+        if not paths:
+            # While we could find an `mzbuild.yml` file for this service, git didn't return any
+            # files that matched this service. That likely means that the directory containing the
+            # mzbuild.yml file has not been added to git; otherwise, the mzbuid.yml file, at a
+            # minimum, would be included in the paths list returned by expand_globs.
+            raise errors.BadSpec(
+                f"{self.image.name} mzbuild exists but its files are unknown to git"
+            )
```

Comment:
Moving discussion from the other thread here. Basically I think we can never hit this case now, modulo programming bugs? So I think either an assert or an error message that captures that it's an internal mzbuild bug is in order?

Reply:
I updated the comment and changed this to an Assertion.